### PR TITLE
[FluentSlider] [FluentNumberField] Fix #2948

### LINKED ISSUE #2609: [Bug] FluentSlider two-way binding issue
https://github.com/microsoft/fluentui-blazor/pull/2591#issuecomment-2324344948

While the mouse wheel is just one example, there are other ways to interact with the slider. For instance, how can I reset the slider value? Fluent UI React Web Component supports this functionality.

https://react.fluentui.dev/?path=/docs/components-slider--docs#controlled

```jsx
import * as React from "react";
import { useId, Button, Label, Slider } from "@fluentui/react-components";
import type { SliderProps } from "@fluentui/react-components";

export const Controlled = () => {
  const id = useId();
  const [sliderValue, setSliderValue] = React.useState(160);
  const onSliderChange: SliderProps["onChange"] = (_, data) =>
    setSliderValue(data.value);
  const resetSlider = () => setSliderValue(0);
  return (
    <>
      <Label htmlFor={id}>
        Control Slider [ Current Value: {sliderValue} ]
      </Label>
      <Slider
        aria-valuetext={`Value is ${sliderValue}`}
        value={sliderValue}
        min={20}
        max={200}
        onChange={onSliderChange}
        id={id}
      />

      <Button onClick={resetSlider}>Reset</Button>
    </>
  );
};

```

React can do this, so how blazor do this?

```razor
<div style="display: flex; flex-direction: column; align-items: flex-start; height: min-content;">
    <label for="rtl-slider" style="margin-left: 8px;">rtl support</label>
    <div dir="rtl" style="width: 100%;">
        <FluentSlider id="rtl-slider" Min="0" Max="100" Step="10" @bind-Value=value2>
            <FluentSliderLabel Position="10">10</FluentSliderLabel>
            <FluentSliderLabel Position="20">20</FluentSliderLabel>
            <FluentSliderLabel Position="40">40</FluentSliderLabel>
            <FluentSliderLabel Position="60">60</FluentSliderLabel>
            <FluentSliderLabel Position="80">80</FluentSliderLabel>
        </FluentSlider>
        
        <FluentButton OnClick="() => value2 = 0 ">Reset</FluentButton>  @* THIS LINE IMPORTANT *@
    </div>
</div>
```

## 🤔 Expected Behavior
I change the value by clicking/dragging, and after I want to reset, then I click `Reset` and the FluentSlider Value should be 0. **That's what true two-way data binding means**

## 😯 Current Behavior
Currently, the value does not reset.

### LINKED ISSUE #2948: fix: desync of FluentNumberField and FluentSlider in 4.10.4
# 🐛 Bug Report

Seeming regressions of #1531 and #1450 

There are 2 Problems

### FluentNumberField

This test demonstrates that the FluentNumberField's two-way data binding is broken in version 4.10.4. This appears to be a regression.

In my web app, I have a component that takes a numeric input and divies a portion to a wallet and a portion to a stash. After a number is selected, a button triggers a command and this command also resets the value of the numeric field to 0.

Unfortunately, the numeric field does not respect the behavior of being reset to 0 by the button command.

Go ahead and select a number like 2, which will enable the reset button. The expected behavior is for the reset button to set the value back to 0, and for the number input field to reflect that value.

You can see in the console logs that the data model's value is reset to 0, however the number field's value is still non-zero both visibly and under the hood. This implies that it isn't respecting the binding.

### FluentSlider

This test demonstrates that the FluentSlider's two-way data binding can desync from its bound value in version 4.10.4. This may be a regression.

While you can see the behavior working for the Reset button, the behavior does not always work for the "Change by 2" button.

Go ahead and set the slider to 3, which will enable the reset button (only enabled at 3). Reset. You see that it works.

Go ahead and set the slider to 1, then keep pressing the "Change by 2" button. In addition to the visual weirdness, by checking the console logs, you will see that it desyncs from the actual value in the data.

When I run it, the console logs always have an accurate data value for the "Change by 2" button's usage. The underlying slider value on blazor side is one step behind. And the visual only updates every 3 clicks of the button.

A second test scenario is: 1. Refresh the page. 2. Set the slider to 1. 3. Click "Change by 2". 4. Click Reset. This will break the visual synchronization with the "Reset" button, which is otherwise usually working correctly.

## 💻 Repro or Code Sample

https://github.com/arazni/blazor-fluent-ui-slider-bug/tree/main

## 🤔 Expected Behavior

When changing a 2-way bound value outside of the component, the visual and component's value should stay synchronized.

## 😯 Current Behavior

There are scenarios where the component does not match the underlying data. See the repro steps on the repo.

## 🔦 Context

Same as  #1531 and #1450, I have buttons that change values nearby fields and sliders that take those changes. See https://arazni.github.io/blades-in-the-sheets/demo for context. You may need to enable sliders over checkboxes in the accessibility tab. Then go back to the demo page, The fund section with the pencil icon enabled is where some of these issues come up.

## 🌍 Your Environment

* At least Windows
* At least Mozilla FireFox
* .NET 9 and Fluent UI Blazor library Version 4.10.4

### PR DESCRIPTION
**Pull request by oneolddev**
# Pull Request

## 📖 Description

Fixes issues reported in #2948. Two issue reported both related to changing the value of the control programatically.

### 🎫 Issues

The underlying issue is that the web components being wrapped were designed to have there states only changed interactively.  `bind-value` does not behave as expected. Inward binding works only on for initialization. Outward binding works as expected.

This gives the appearance of a *sync* issue.

## 👩‍💻 Reviewer Notes

`FluentSlider` - #2665 did not fully fix #2609. The original fix was to prevent the [flickering when the slider was moved rapidly](https://github.com/microsoft/fluentui-blazor/issues/2609#issuecomment-2326339496) originally reported. I was not able to reproduce this problem with the current implementation of `FluentDataGrid`. I did end up adding debouncing with the `DebounceAction`.  I believe I did this correctly.

`FluentNumberField` - The fix was to ensure the underlying web component reflected the correct value. I could not find a way of distinguishing if the value was changed by manipulating the control or if it was changed programatically. I opted to use some javascript to force the value in the web component. I also added a `DisposeAsync`. It was missing.

I've also included an `issue-tester` page as part of this draft

### General

- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/microsoft/fluentui-blazor/blob/master/docs/contributing.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new component
- [ ] I have modified an existing component
- [x] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

File: examples/Demo/Shared/Pages/Lab/IssueTesterPage.razor
Comment (vnbaaij) [inline examples/Demo/Shared/Pages/Lab/IssueTesterPage.razor:1]: For now this is fine but we have the  lssueTesterPage as the 'container' (so you do not need to mess with the @page and title, etc.) and the IssueTester as the actual pungent to put the code in.

**Review by vnbaaij — COMMENTED** [review]
Will test this soop

File: examples/Demo/Shared/Pages/Lab/IssueTesterPage.razor
Comment (oneolddev) [reply examples/Demo/Shared/Pages/Lab/IssueTesterPage.razor:1]: I left it in for testing purposes only. was planning to remove it 😄

**Comment by oneolddev** [comment]
@vnbaaij 

I've fixed the unit test by easing the BUnit JSInterop mode. I'm not sure if that breaks any testing conventions.

Should this be split into two PRs? One for `FluentSlider` and one for `FluentNumberField`.

**Comment by oneolddev** [comment]
Found another issue with my fix for `FluentNumberField` 😞

File: src/Core/Components/NumberField/FluentNumberField.razor.cs [hunk 1]
```diff
@@ -173,5 +184,34 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
             }
 
         }
+        else
```
Comment (dvoituron) [inline src/Core/Components/NumberField/FluentNumberField.razor.cs:187]: I don't like calling a JS method every time a render is raised :-(

Why not use a JS Observable to apply the `value` attribute to the `value` JS property?
Like this: https://github.com/microsoft/fluentui-blazor/blob/dev-v5/src/Core/Components/TextInput/FluentTextInput.razor.ts

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/Core/Components/NumberField/FluentNumberField.razor.cs (hunk 1 above)
Comment (oneolddev) [reply src/Core/Components/NumberField/FluentNumberField.razor.cs:187]: I'll take a look at that approach 🙏

Comment (oneolddev) [reply src/Core/Components/NumberField/FluentNumberField.razor.cs]: @dvoituron 

This was a great suggestion 😄.  It's definitely better than my original approach.

**Review by dvoituron — APPROVED** [review]
Next time, please create one PR to fix one problem. This makes it easier to maintain the code.
Thank you :-)

**Review by vnbaaij — APPROVED** [review]

**Pull request merged**